site/src/pages/WorkspaceSettingsPage/WorkspaceParametersPage/WorkspaceParametersPageViewExperimental.tsx

```diff
@@ -80,6 +71,29 @@ export const WorkspaceParametersPageViewExperimental: FC<
 		workspace.template_require_active_version &&
 		!canChangeVersions;
 
+	const setFormFieldTouched = useCallback(
+		(parameter: PreviewParameter) => {
+			const values = form.values.rich_parameter_values ?? [];
+			const index = values.findIndex((p) => p.name === parameter.name);
+			if (index !== -1) {
+				const path = `rich_parameter_values.${index}.value`;
+				if (getIn(form.touched, path) !== true) {
+					form.setFieldTouched(path, true, false);
+				}
+			}
+		},
+		[form.touched, form.setFieldTouched, form.values.rich_parameter_values],
+	);
+
+	useEffect(() => {
+		parameters.forEach((parameter) => {
+			if (autofillByName[parameter.name] === undefined) {
+				return;
+			}
+			setFormFieldTouched(parameter);
+		});
+	}, [autofillByName, parameters, setFormFieldTouched]);
```

Comment:
this is really not an appropriate use of an effect.

every time the parameters change, you're iterating over all of them to see if they exist in `autofillByName`, and immediately causing a new render by updating state.